assets/src/higher-order-components/base-control.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * WordPress Imports
+ */
+import { Component, createHigherOrderComponent, compose } from '@wordpress/element';
+import { BaseControl, withInstanceId } from '@wordpress/components';
+
+/**
+ * External Imports
+ */
+import { PropTypes } from 'prop-types';
+
+export default ( customId = '' ) => createHigherOrderComponent(
+	compose( [
+		withInstanceId,
+		( WrappedComponent ) => {
+			return class extends Component {
+				static propTypes = {
+					label: PropTypes.string,
+					instanceId: PropTypes.oneOfType( [
+						PropTypes.number,
+						PropTypes.string,
+					] ),
+					className: PropTypes.string,
+					help: PropTypes.string,
+				};
+
+				render() {
+					const {
+						label,
+						instanceId,
+						className,
+						help,
+					} = this.props;
+					const id = `inspector-${ customId }-control-${ instanceId }`;
```

Comment:
where is `customId ` coming from? 
ie: why is it not in the list of constants extracted from the props like `instanceId` ?

Reply:
It's coming from the argument passed into the HOC: `( customId = '' ) =>`

The signature of the HOC is `withBaseControl( customId = '' )( WrappedComponent)`

Reply:
ya I saw that it was an argument passed to the HOC but didn't realize that that meant it was automatically passed into the `render()` method, ie: I thought all incoming arguments just became part of the props and that you still had to extract them within each method.

so kewl... that's two new things I just learned 👍

Reply:
Well its not really "passed" in.  It's just that the argument is within scope of all the code contained within the hoc function. It's similar to if you did something like this:

```
const something = ( someArg ) => {
    return function() {
         console.log(someArg);
    }
};

something('hello there')();
//would print 'hello there'
```